bundles/framework/rpc/service/RpcService.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @class Oskari.mapframework.bundle.rpc.service.RpcService
+ * Allows extend RPC supported funtions
+ *
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.rpc.service.RpcService',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     *
+     */
+
+    function (instance) {
+        this.instance = instance;
+        this.conf = this.instance.conf || {};
+        this.channel = this.instance._channel;
+        this._log = Oskari.log('RpcService');
+        this._availableFunctions = {};
+        this._allowedFunctions = this.conf.allowedfunctions || [];
+        this._allowedFunctionsFromConf = Array.isArray(this.conf.allowedfunctions);
+    }, {
+        __name: 'Rpc.RpcService',
+        __qname: 'Oskari.mapframework.bundle.rpc.service.RpcService',
+
+        getQName: function () {
+            return this.__qname;
+        },
+
+        getName: function () {
+            return this.__name;
+        },
+
+        /**
+         * @method init
+         * Initializes the service
+         */
+        init: function () {
+        },
+
+        /**
+         * @method addFunction
+         * Add function to RPC
+         *
+         * @param func function that should be added
+         */
+        addFunction: function (func) {
```

Comment:
Let's change this so we give the function name as string. Just to make sure the build doesn't mangle it up (if it doesn't now it might after some future update).
```suggestion
        addFunction: function (name, func) {
```
This would change the example to:
```
    rpcService.addFunction('example', function () {
        console.log('New added RPC function');
        return 'example';
    });

Reply:
Changed